airbyte-integrations/bases/bases-destination-jdbc/src/main/java/io/airbyte/integrations/destination/staging/GeneralStagingFunctions.java

```diff
@@ -86,6 +91,37 @@ public static void copyIntoTableFromStage(final JdbcDatabase database,
     }
   }
 
+  private static CompletableFuture<Optional<Exception>> dropStageTask(final WriteConfig writeConfig,
+                                                                      final StagingOperations stagingOperations,
+                                                                      final boolean purgeStagingData,
+                                                                      final JdbcDatabase database) {
+    return CompletableFuture.supplyAsync(() -> {
```

Comment:
how many threads does this use? javadoc says it depends on ForkJoinPool.commonPool, which google suggests is limited to `cpu_count - 1` threads, which is probably pretty restrictive on cloud (where we set cpu=3 or something for snowflake?)

I suspect we need to supply our own thread pool. dbt defaults to 4 threads, but we can probably go even higher

Reply:
I also like the idea of us providing the number explicitly.  e.g. in the future, this number should scale with the size of your compute warehouse.  For now, a contestant (4?) is probably a good stating point